packages/assets-controllers/src/TokenListController.ts

```diff
@@ -204,104 +204,110 @@ export class TokenListController extends StaticIntervalPollingController<TokenLi
             tokenList: this.state.tokensChainsCache[this.chainId]?.data || {},
           };
         });
```

Comment:
We discussed it's probably a good idea to remove entries when their network is deleted.  But I don't think it's necessary to do that now.  

Untested example:
```ts
    const chainIdsToRemove = [] as Hex[];
    for (const chainId in Object.keys(this.state.tokensChainsCache)) {
      if (!networkControllerState.networkConfigurationsByChainId[chainId as Hex]) {
        chainIdsToRemove.push(chainId as Hex)
      }
    }
    if (chainIdsToRemove.length > 0) {
      this.update(() => {
        return {
          ...this.state,
          tokensChainsCache: chainIdsToRemove.reduce((acc, chainId) => {
            delete acc[chainId]
            return acc
          }, this.state.tokensChainsCache)
        }
      })
    }
```